src/main/java/com/ciandt/techgallery/service/impl/GooglePlusCommunicationServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+    aclEntries.add(resource);
+
+    Acl acl = new Acl();
+    acl.setItems(aclEntries);
+    acl.setDomainRestricted(true); // Required, this does the domain restriction
+
+    // Create a new activity object to be executed
+    String content = createContent(feature, currentUserMail, endorsedMail, technologyName, score, comment);
+    Activity activity = new Activity().setObject(new Activity.PlusDomainsObject().setOriginalContent(content))
+        .setAccess(acl);
+
+    // Attach the link
+    Activity.PlusDomainsObject.Attachments attachment = new Activity.PlusDomainsObject.Attachments();
+    attachment.setObjectType("article");
+    attachment.setUrl(techGalleryLink);
+    List<Activity.PlusDomainsObject.Attachments> attachments = new ArrayList();
+    attachments.add(attachment); // You can also add multiple attachments to the
+                                 // post
+    activity.getObject().setAttachments(attachments);
+
+    // Creating a google credential in base of the header authorization
+    String header = req.getHeader("Authorization");
+    String accesstoken = header.substring(header.indexOf(' ')).trim();
+    GoogleCredential credential = new GoogleCredential().setAccessToken(accesstoken);
+
+    // Create a new authorized API client according the credential
+    PlusDomains plusDomains = new PlusDomains.Builder(new NetHttpTransport(), new JacksonFactory(), credential).build();
+
+    // Execute the API request, which calls `activities.insert` for the logged
+    // in user
+    activity = plusDomains.activities().insert("me", activity).execute();
+  }
+
+  /**
+   * This method confirm the user preference to post on your Google+ profile
+   * 
+   * @param user
+   *          the user logged in
+   * @throws NotFoundException
+   * @throws BadRequestException
+   * @throws InternalServerErrorException
+   */
+  private void verifyRequirements(User user)
+      throws NotFoundException, BadRequestException, InternalServerErrorException {
+    final TechGalleryUser techUser = userService.getUserByGoogleId(user.getUserId());
+    if (techUser.getPostGooglePlusPreference().equals(Boolean.FALSE)) {
+      throw new BadRequestException(ValidationMessageEnums.NOT_PERMITTED_BY_USER.message());
+    }
+  }
+
+  /**
+   * Method that create a content for the post on Google+
+   * 
+   * @param feature
+   *          the type of feature performed by user.
+   * @param currentUserMail
+   *          is the email of the user logged in.
+   * @param endorsedMail
+   *          is the email of the endorsed user in case of endorse feature.
+   * @param technologyName
+   *          is the name of technology performed by feature.
+   * @param score
+   *          is the positive or negative recommendation in case of
+   *          recommendation feature.
+   * @return
+   */
+  private String createContent(FeatureEnum feature, String currentUserMail, String endorsedMail, String technologyName,
```

Comment:
@tsr-thulio If you put this method in the FeatureEnum it would be better. If you do that it wouldn't need to do the switch case (usually this is a smell that OO should be applied).